cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/analyzer/LowResourceUtilizationTest.java

```diff
@@ -0,0 +1,186 @@
+    goalConfigs.put(AnalyzerConfig.NETWORK_INBOUND_LOW_UTILIZATION_THRESHOLD_CONFIG, 0.14445);
+    p.add(params(clusterModel, new NetworkInboundUsageDistributionGoal(), goalConfigs, false));
+
+
+    // Expect rebalance on network outbound usage distribution goal since one broker's network outbound usage is
+    // above the low network outbound utilization threshold
+    clusterModel = createSmallClusterModel();
+    goalConfigs = getDefaultGoalConfigs();
+    goalConfigs.put(AnalyzerConfig.NETWORK_OUTBOUND_LOW_UTILIZATION_THRESHOLD_CONFIG, 0.14);
+    p.add(params(clusterModel, new NetworkOutboundUsageDistributionGoal(), goalConfigs, true));
+
+    // Expect no rebalance on network outbound usage distribution goal since all brokers' network outbound usage is
+    // below the low network outbound utilization threshold
+    clusterModel = createSmallClusterModel();
+    goalConfigs = getDefaultGoalConfigs();
+    goalConfigs.put(AnalyzerConfig.NETWORK_OUTBOUND_LOW_UTILIZATION_THRESHOLD_CONFIG, 0.1639);
+    p.add(params(clusterModel, new NetworkOutboundUsageDistributionGoal(), goalConfigs, false));
+
+    return p;
+  }
+
+  private static Map<String, Object> getDefaultGoalConfigs() {
+    return new HashMap<String, Object>() {{
+      put(MonitorConfig.BOOTSTRAP_SERVERS_CONFIG, "bootstrap.servers");
+      put(ExecutorConfig.ZOOKEEPER_CONNECT_CONFIG, "connect:1234");
+    }};
+  }
+
+  /**
+   * Creates a small cluster model with broker resource utilization percentage shown below:
+   *    Broker-0:
+   *      CPU:    34.75%
+   *      Disk:   28%
+   *      NW_IN:  13%
+   *      NW_OUT: 14.75%
+   *
+   *    Broker-1:
+   *      CPU:    14%
+   *      Disk:   15.5%
+   *      NW_IN:  7%
+   *      NW_OUT: 5.8%
+   *
+   *    Broker-2:
+   *      CPU:    9.75%
+   *      Disk:   13.5%
+   *      NW_IN:  6.5%
+   *      NW_OUT: 0%
+   *
+   * @return cluster model
+   */
+  private static ClusterModel createSmallClusterModel() {
+    return DeterministicCluster.smallClusterModel(BROKER_CAPACITY);
+  }
+
+  private static Object[] params(ClusterModel clusterModel,
+                                 ResourceDistributionGoal resourceDistributionGoal,
+                                 Map<String, Object> goalConfigs,
+                                 Boolean expectRebalance) {
+    return new Object[]{clusterModel, resourceDistributionGoal, goalConfigs, expectRebalance};
+  }
+
+  @Test
+  public void test() throws OptimizationFailureException {
+    Map<TopicPartition, List<ReplicaPlacementInfo>> initReplicaDistribution = _clusterModel.getReplicaDistribution();
+    Map<TopicPartition, ReplicaPlacementInfo> initLeaderDistribution = _clusterModel.getLeaderDistribution();
+
+    _resourceDistributionGoal.configure(_goalConfigs);
+    assertTrue("Failed to optimize " + _resourceDistributionGoal.name(),
+        _resourceDistributionGoal.optimize(_clusterModel, Collections.emptySet(), new OptimizationOptions(Collections.emptySet())));
```

Comment:
We cannot do that since `_resourceDistributionGoal.optimize(_clusterModel, ..)` changed the `_clusterModel` and the values of `initReplicaDistribution` and `initLeaderDistribution` would be different if we calculate them after the optimization